plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -888,6 +890,240 @@ public void testSystemSchemas()
+                .append("parent", new Document().append("creator", dbRef).append("child", objectId));
+
+        client.getDatabase("test").getCollection(tableName).insertOne(document);
+
+        assertThat(query("SELECT parent.child, parent.creator.databaseName, parent.creator.collectionName, parent.creator.id FROM test." + tableName))
+                .matches("SELECT " + expectedValue + ", varchar 'test', varchar 'creators', " + expectedValue)
+                .isFullyPushedDown();
+        assertQuery(
+                "SELECT typeof(parent.creator) FROM test." + tableName,
+                "SELECT 'row(databaseName varchar, collectionName varchar, id " + expectedType + ")'");
+
+        assertUpdate("DROP TABLE test." + tableName);
+    }
+
+    @Test(dataProvider = "dbRefProvider")
+    public void testProjectionPushdownWithPredefinedDBRefKeyword(Object objectId, String expectedValue, String expectedType)
+    {
+        String tableName = "test_projection_pushdown_with_predefined_dbref_keyword_" + randomNameSuffix();
+
+        DBRef dbRef = new DBRef("test", "creators", objectId);
+        Document document = new Document()
+                .append("_id", new ObjectId("5126bbf64aed4daf9e2ab771"))
+                .append("col1", "foo")
+                .append("parent", new Document().append("id", dbRef));
+
+        client.getDatabase("test").getCollection(tableName).insertOne(document);
+
+        assertThat(query("SELECT parent.id, parent.id.id FROM test." + tableName))
+                .skippingTypesCheck()
+                .matches("SELECT row('test', 'creators', %1$s), %1$s".formatted(expectedValue))
+                .isFullyPushedDown();
+        assertQuery(
+                "SELECT typeof(parent.id), typeof(parent.id.id) FROM test." + tableName,
+                "SELECT 'row(databaseName varchar, collectionName varchar, id %1$s)', '%1$s'".formatted(expectedType));
+
+        assertUpdate("DROP TABLE test." + tableName);
+    }
+
+    @Test(dataProvider = "dbRefProvider")
+    public void testProjectionPushdownWithSameFieldAsDBRef(Object objectId, String expectedValue, String expectedType)
+    {
+        String tableName = "test_projection_pushdown_with_same_field_as_dbref_" + randomNameSuffix();
+
+        Document document = new Document()
+                .append("creator", new Document().append("databaseName", "test").append("collectionName", "creators").append("id", objectId));
+
+        client.getDatabase("test").getCollection(tableName).insertOne(document);
+
+        assertThat(query("SELECT creator.databaseName, creator.collectionName, creator.id FROM test." + tableName))
+                .matches("SELECT varchar 'test', varchar 'creators', " + expectedValue)
+                .isFullyPushedDown();
+        assertQuery(
+                "SELECT typeof(creator) FROM test." + tableName,
+                "SELECT 'row(databaseName varchar, collectionName varchar, id " + expectedType + ")'");
```

Comment:
`projection_pushdown_enabled` session property affects the final output. Perhaps, this should throw an exception instead. 
```sql
trino:tpch> select * from test;
           _id            |                                  creator
--------------------------+----------------------------------------------------------------------------
 5126bbf64aed4daf9e2ab772 | {collectionName=creators, id=5126bc054aed4daf9e2ab772, databaseName=users}
 5126bbf64aed4daf9e2ab771 | {collectionName=users, id=creators, databaseName=5126bc054aed4daf9e2ab772}

trino:tpch> set session mongodb.projection_pushdown_enabled = true;
trino:tpch> select creator.id from test;
            id
--------------------------
 5126bc054aed4daf9e2ab772
 NULL

trino:tpch> set session mongodb.projection_pushdown_enabled = false;
trino:tpch> select creator.id from test;
            id
--------------------------
 5126bc054aed4daf9e2ab772
 creators
```

Reply:
How did you insert data into the table? via trino or using mongoclient?

is `creator` a Document or DBRef column? what is the type of `id` column here?

Reply:
```java
    @Test
    public void test()
    {
        String tableName = "test_" + randomNameSuffix();

        Document document1 = new Document()
                .append("_id",  new ObjectId("5126bbf64aed4daf9e2ab772"))
                .append("creator", new Document().append("collectionName", "collection").append("id", "2").append("databaseName", "db"));
        Document document2 = new Document()
                .append("_id",  new ObjectId("5126bbf64aed4daf9e2ab771"))
                .append("creator", new DBRef("db", "collection", "1"));
        client.getDatabase("test").getCollection(tableName).insertOne(document1);
        client.getDatabase("test").getCollection(tableName).insertOne(document2);

        assertThat(query("SELECT creator.id FROM test." + tableName))
                .skippingTypesCheck()
                .matches("VALUES ('1'), (NULL)");
    }

```